File: rental-backend/src/main/java/com/homeflex/features/vehicle/dto/response/VehicleResponse.java
package com.homeflex.features.vehicle.dto.response;

import com.homeflex.core.dto.response.UserDto;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record VehicleResponse(
        UUID id,
        UUID ownerId,
        String brand,
        String model,
        Integer year,
        String transmission,
        String fuelType,
        BigDecimal dailyPrice,
        String currency,
        String status,
        String description,
        Integer mileage,
        Integer seats,
        String color,
        String licensePlate,
        String pickupCity,
        String pickupAddress,
        int viewCount,
        List<VehicleImageDto> images,
        UserDto owner,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {}

File: rental-app-flutter/lib/features/vehicles/models/vehicle_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'vehicle_image_dto.dart';

part 'vehicle_response.freezed.dart';
part 'vehicle_response.g.dart';

@freezed
abstract class VehicleResponse with _$VehicleResponse {
  const factory VehicleResponse({
    required String id,
    required String ownerId,
    required String brand,
    required String model,
    required int year,
    required String transmission,
    required String fuelType,
    required double dailyPrice,
    required String currency,
    required String status,
    String? description,
    int? mileage,
    int? seats,
    String? color,
    String? licensePlate,
    String? pickupCity,
    String? pickupAddress,
    required int viewCount,
    List<VehicleImageDto>? images,
    required String createdAt,
    required String updatedAt,
  }) = _VehicleResponse;

  factory VehicleResponse.fromJson(Map<String, dynamic> json) => _$VehicleResponseFromJson(json);
}
